vaccine_feed_ingest/utils/normalize.py

```diff
@@ -262,6 +263,67 @@ def normalize_phone(
     return contacts
 
 
+def parse_address(address: str) -> OrderedDict[str, str]:
+    """Parse address, but may require subsequent cleaning depending on dataset."""
+
+    address = address.replace("\n", ", ")
+
+    parsed_address, address_type = usaddress.tag(address)
+    if address_type != "Street Address":
+        logger.warning(
+            f"Couldn't normalize address '{address}' of type {address_type}; best guess: {parsed_address}"
+        )
+
+    # Fixup: At least one address has "WI, USA" in the "StateName" component.
+    # Strip non-state components
+    if parsed_address.get("StateName"):
+        parsed_address["StateName"] = parsed_address["StateName"].partition(",")[0]
+
+    return parsed_address
+
+
+def normalize_address(patched_address: OrderedDict[str, str]) -> location.Address:
+    """Normalize a usaddress-parsed (and possibly -patched) address.
+
+    Calling convention:
+
+        parsed_address, address_type = parse_address("1600 Pennsylvania Ave NW, Washington DC 20500")
+        patched_address = apply_local_fixes(parsed_address)
+        address = normalize_address(patched_address)
+
+
+    where apply_local_fixes does any cleanup fixes needed for the dataset.
+    """
+
+    address_kwargs = {
+        # "street1",
+        # "city",
+        # "state",
+        # "zip"
+    }
+    street_buffer: List[str] = []
+    suite_buffer: List[str] = []
+    while len(patched_address) > 0:
```

Comment:
Is there a minimum set of fields that we want to _require_ for this to work without erroring? On the one hand it seems good to be as flexible as possible, on the other hand I want to guard against folks not following the expected convention for calling this.

Maybe we should consider setting an extra field in `parse` that we then require in `normalize`?

Reply:
Hmm, that feels like a design smell to me --- and I'd expect anyone who was trying to work around the convention would read the code to see what needs to be in `patched_address` and add it, so I don't know that it buys much?  I don't have strong feelings either way, **shrug**.

Reply:
I think you're right!